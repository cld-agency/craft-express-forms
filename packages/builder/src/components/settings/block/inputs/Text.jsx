import React from 'react';
import BaseInput from './BaseInput';

class Text extends BaseInput {
  renderInput() {
    const { name, value, saveHandler, valueToCamelCase = false } = this.props;

    return (
      <input
        className="text fullwidth"
        type="text"
        name={name}
        value={value}
        onChange={saveHandler}
        data-value-to-camel-case={valueToCamelCase ? '1' : ''}
      />
    );
  }
}

export default Text;
