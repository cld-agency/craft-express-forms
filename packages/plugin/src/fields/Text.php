<?php

namespace Solspace\ExpressForms\fields;

class Text extends BaseField
{
    public function getType(): string
    {
        return self::TYPE_TEXT;
    }
}
