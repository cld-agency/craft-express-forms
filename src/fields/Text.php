<?php

namespace Solspace\ExpressForms\fields;

class Text extends BaseField
{
    /**
     * @return string
     */
    public function getType(): string
    {
        return self::TYPE_TEXT;
    }
}
