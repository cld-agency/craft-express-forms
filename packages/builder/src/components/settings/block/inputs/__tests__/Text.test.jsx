import { shallow } from 'enzyme';
import React from 'react';
import Text from '../Text';

describe('<Text />', () => {
  let wrapper;
  let onChange;
  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(<Text label="A Test Input" name="test_input" value="a value" saveHandler={onChange} />);
  });

  it('renders & displays correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains a text input', () => {
    expect(wrapper.find("input[name='test_input']")).toHaveLength(1);
  });

  it('changes value', () => {
    const value = 'A different value';
    wrapper.find("input[name='test_input']").simulate('change', { target: { value } });
    expect(onChange).toBeCalledWith({ target: { value: 'A different value' } });
  });
});
