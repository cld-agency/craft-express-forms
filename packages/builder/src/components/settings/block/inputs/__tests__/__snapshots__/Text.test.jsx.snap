// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders & displays correctly 1`] = `
<div
  className="block-row"
>
  <div
    className="label"
  >
    <WithStyles(ForwardRef(Tooltip))
      placement="bottom"
      title=""
    >
      <label
        className=""
      >
        A Test Input
      </label>
    </WithStyles(ForwardRef(Tooltip))>
  </div>
  <div
    className="input ltr"
  >
    <input
      className="text fullwidth"
      data-value-to-camel-case=""
      name="test_input"
      onChange={[MockFunction]}
      type="text"
      value="a value"
    />
  </div>
</div>
`;
